core/services/functions/connector_handler.go

```diff
@@ -47,3 +83,105 @@ func (h *functionsConnectorHandler) Start(ctx context.Context) error {
 func (h *functionsConnectorHandler) Close() error {
 	return nil
 }
+
+func (h *functionsConnectorHandler) handleSecretsList(ctx context.Context, gatewayId string, msg *api.Message, fromAddr ethCommon.Address) {
+	type ListRow struct {
+		SlotID     uint   `json:"slot_id"`
+		Version    uint64 `json:"version"`
+		Expiration int64  `json:"expiration"`
+	}
+
+	type ListResponse struct {
+		Success bool      `json:"success"`
+		Error   string    `json:"error,omitempty"`
+		Rows    []ListRow `json:"rows,omitempty"`
+	}
+
+	var response ListResponse
+	snapshot, err := h.storage.List(ctx, fromAddr)
+	if err == nil {
+		response.Success = true
+		response.Rows = make([]ListRow, len(snapshot))
+		for i, row := range snapshot {
+			response.Rows[i] = ListRow{
+				SlotID:     row.SlotId,
+				Version:    row.Version,
+				Expiration: row.Expiration,
+			}
+		}
+	} else {
+		response.Error = fmt.Sprintf("Failed to list secrets: %v", err)
```

Comment:
We don't need to set Success = false, because it is false by default.
Helpers make little sense, since I have the single response path. The flow seems clear - we construct the response (whatever it is success or failure) and send.